prov/verbs/src/fi_verbs.c

```diff
@@ -663,14 +644,38 @@ static int fi_ibv_getinfo(uint32_t version, const char *node, const char *servic
 			  uint64_t flags, struct fi_info *hints, struct fi_info **info)
 {
 	struct rdma_cm_id *id;
+	struct rdma_addrinfo *rai;
+	struct fi_info *fi;
 	int ret;
 
-	ret = fi_ibv_getepinfo(node, service, flags, hints, info, &id);
+	ret = fi_ibv_create_ep(node, service, flags, hints, &rai, &id);
 	if (ret)
 		return ret;
 
+	if (!(fi = fi_allocinfo())) {
+		ret = -FI_ENOMEM;
+		goto err;
+	}
+
+	ret = fi_ibv_rai_to_fi(rai, fi);
+	if (ret)
+		goto err;
+
+	ret = fi_ibv_fill_info_attr(id->verbs, hints, fi);
+	if (ret)
+		goto err;
+
+	*info = fi;
+
 	rdma_destroy_ep(id);
+	rdma_freeaddrinfo(rai);
 	return 0;
+err:
+	if (fi)
+		fi_freeinfo(fi);
```

Comment:
> OK, I will create another patch to address this. Are we avoiding branches
> here for performance reasons?

No - the performance doesn't matter here.  It's more of a style issue.  There's no reason to jump to a cleanup point that we know should not execute.  That usually forces early initialization of variables so that branches will be skipped.